include/deal.II/matrix_free/fe_evaluation_data.h

```diff
@@ -1183,7 +1184,13 @@ FEEvaluationData<dim, Number, is_face>::set_data_pointers(
      n_quadrature_points);
 
   const unsigned int allocated_size = size_scratch_data + size_data_arrays;
+#  ifdef DEBUG
+  scratch_data_array->clear();
+  scratch_data_array->resize(allocated_size,
+                             Number(numbers::signaling_nan<double>()));
```

Comment:
Should this be the following instead?
```suggestion
                            numbers::signaling_nan<Number>());
```

Reply:
Won't work because this will be instantiated for Number=VectorizedArray<float> and <double>.
I was too lazy to extend signaling_nan.

Reply:
Ah, but then what are the semantics of casting a signaling nan `double` to a `VectorizedArray`?

Leave it as is for the moment -- the patch isn't going to get merged for now while there are failures, but if you don't mind, add the specializations of `signaling_nan` for `VectorizedArray` objects.

Reply:
It is calling the constructor that puts this value into all values:
https://github.com/dealii/dealii/blob/cb8330216531c7c32979746ec815b21c986cdc70/include/deal.II/base/vectorization.h#L440

Reply:
But you should be able to use
```suggestion
                             Number(numbers::signaling_nan<ScalarNumber>()));
```
which is defined here:
https://github.com/dealii/dealii/blob/b8f23f11a6ab25f16c65b56be56f216421717ff7/include/deal.II/matrix_free/fe_evaluation_data.h#L125-L126